File: UI/task-ui/src/app/Service/ITaskService.ts
import { TaskModel } from 'src/app/model/taskModel';
import { Observable } from 'rxjs/internal/Observable';
import { ParentTaskModel } from 'src/app/model/parentTaskModel';
import { TaskQueryModel } from 'src/app/model/taskQueryModel';

export abstract class ITaskService {
    abstract getAllTask(): Observable<TaskModel[]>;
    abstract getTaskById(id: number): Observable<TaskModel>;
    abstract queryTasks(query: TaskQueryModel): Observable<TaskModel[]>;
    abstract getAllParentTask(): Observable<ParentTaskModel[]>;
    abstract AddTaks(task: TaskModel);
    abstract UpdateTaks(task: TaskModel):Observable<any>;
    abstract DeleteTaks(task: TaskModel):Observable<any>;
    abstract CloseTask(task: TaskModel):Observable<any>;
}

File: Service/Task.BusinessLayer/ITaskService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Task.Entities;

namespace Task.BusinessLayer
{
    public interface ITaskService:IDisposable
    {
        
        TaskModel GetTaskById(int id);
        TaskModel UpdateTaks(TaskModel id);

        TaskModel AddTask(TaskModel task);

        bool DeleteTaks(TaskModel id);

        ICollection<TaskModel> GetAllTasks();

        //ICollection<TaskModel> QueryTask(string name,DateTime? startDate,DateTime? endDate,int? priority, string parentTask);

        ICollection<TaskModel> QueryTask(TaskQueryModel query);

        ICollection<ParentTaskModel> GetAllParentTasks();

        TaskModel CloseTask(TaskModel task);
    }
}
